File: tests/exercises/division/evaluation/Evaluator.cs
using System;
using Tested;

public class Evaluator {
    public static EvaluationResult Evaluate(object? actual) {
        if (actual is DivideByZeroException) {
            return new EvaluationResult(true, actual.ToString(), actual.ToString());
        } else {
            var messages = new List<Message>() { new Message("Expected DivideByZeroException, got something else.") };
            return new EvaluationResult(false, "System.DivideByZeroException", actual == null ? "" : actual.ToString(), messages);
        }
    }

    public static EvaluationResult Runtime(object? actual) {
        throw new ArgumentOutOfRangeException("hello");
    }
}

File: tests/exercises/division/evaluation/Evaluator.cpp
#include <exception>
#include "evaluation_result.h"

class Evaluator {
public:
    static EvaluationResult evaluate(const std::exception_ptr &actual) {
        try {
            if(actual) {
                std::rethrow_exception(actual);
            }
        } catch (const std::logic_error& e) {
            return EvaluationResult(true, e.what(), e.what());
        } catch (...) {
            // do nothing
        }

        return EvaluationResult(false, "std::logic_error", "something else", {
            Message("Expected std::logic_error, got something else.")
        });
    }

    static EvaluationResult runtime(const std::exception_ptr &actual) {
        throw std::out_of_range("hello");
    }
};
